### PR TITLE
External app namespaces: interface update and input validation

### PR DESCRIPTION
**Pull request by jekloudaMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

Mark composeExtension propery on IQueryMessageExtensionResponse as optional and add input validation for external app APIs.

### Main changes in the PR:

1. `composeExtension` is now an optional property on `IQueryMessageExtensionResponse`
2. Added input validation for `IOriginalRequest` and app ID inputs

### Validation performed:

1. Unit tests added

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

Yes

### End-to-end tests added:

No

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

Yes

### CODE REVIEW

File: change/@microsoft-teams-js-ff815d37-beb3-4aba-a561-f6802208de9d.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Marked `composeExtension` property on `IQueryMessageExtensionResponse` as optional",
```
Comment (erinha) [inline change/@microsoft-teams-js-ff815d37-beb3-4aba-a561-f6802208de9d.json:3]: Should this also include the additional input validation and removal of `cardActionsConfig`?

File: packages/teams-js/test/internal/utils.spec.ts [hunk 2]
```diff
@@ -276,4 +277,21 @@ describe('utils', () => {
       }
     });
   });
+
+  describe('validateAppIdIsGuid', () => {
```
Comment (erinha) [inline packages/teams-js/test/internal/utils.spec.ts:281]: I don't think this matches exactly the set of unit tests we added in the host SDK. Is that intentional?

File: change/@microsoft-teams-js-ff815d37-beb3-4aba-a561-f6802208de9d.json (hunk 1 above)
Comment (jekloudaMSFT) [reply change/@microsoft-teams-js-ff815d37-beb3-4aba-a561-f6802208de9d.json:3]: `cardActionsConfig` was a separate PR. I'll mention the input validation

File: packages/teams-js/test/internal/utils.spec.ts (hunk 2 above)
Comment (jekloudaMSFT) [reply packages/teams-js/test/internal/utils.spec.ts:281]: There was a custom error test that doesn't apply here, but I added the "valid app ID, invalid v4 uuid" test

File: change/@microsoft-teams-js-ff815d37-beb3-4aba-a561-f6802208de9d.json (hunk 1 above)
Comment (erinha) [inline change/@microsoft-teams-js-ff815d37-beb3-4aba-a561-f6802208de9d.json:3]:
```suggestion
  "comment": "Marked `composeExtension` property on `IQueryMessageExtensionResponse` as optional and added input validation to `externalAppAuthentication` and `externalAppCardActions` capabilities.",
```

**Review by erinha — APPROVED** [review]

**Review by jadahiya-MSFT — APPROVED** [review]

**Pull request merged**